[ki1r0] iP

ki1r0 opened the pull request:
# 📌 Judy - Your Ultimate Task Manager

> "The only way to do great work is to love what you do." – Steve Jobs

Judy frees your mind from remembering tasks so you can focus on what truly matters.

## 🚀 Why Choose Judy?
- 📝 **Text-based** – Simple and intuitive  
- ⚡ **Super fast** – No lag, just instant productivity  
- 🎯 **Easy to learn** – Get started in seconds  

## 🛠 Features
1. ✔ **Managing tasks** – Add, remove, and organize tasks effortlessly  
2. ⏳ **Managing deadlines** *(coming soon)*  
3. ⏰ **Reminders** *(coming soon)*  

## 📥 How to Get Started
1. [Download Judy from GitHub](https://github.com/your-repo/judy).
2. **Double-click** to run it.
3. Add your tasks using `todo`, `deadline`, or `event` commands.
4. Let Judy handle the rest! 🚀

## ✅ To-Do List
- [x] Support adding tasks  
- [x] Instant task management  
- [ ] Implement deadlines *(coming soon)*  
- [ ] Add reminder notifications *(coming soon)*  

## 🖥 Code Example
```java
public class Judy {
    public static void main(String[] args) {
        System.out.println("Welcome to Judy - Your Ultimate Task Manager!");
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
level 0
rename, greet, exit

Commit:
no message

Commit:
Echo

Commit:
add, list

Commit:
Mark as Done, task class

Commit:
ToDo, Event, Deadline

Commit:
no message

Commit:
Complete class Event, Deadline, Judy, Task, Todo

Commit:
Automated Text UI Testing

Commit:
Error Handling

Commit:
Delete

Commit:
Use Enums

Commit:
Save and Load

Commit:
Date and Times

Commit:
to avoid fast forwarding

Commit:
Revert "Use Enums"

This reverts commit 5084aee42f0df7eeda95a96cfa94f2982ccb4448.

Commit:
Merge branch 'branch-Level-7' into 'master'

Commit:
merge resolved

Commit:
merge resolved

Commit:
more oop

Commit:
Merge branch 'branch-Level-8'

# Conflicts:
#	data/judy.txt
#	out/production/ip-master/Task.class
conflict resolved

Commit:
more OOP

Commit:
more OOP

Commit:
More OOP:Add Parser, Command

Commit:
Organize into Packages

Commit:
Merge remote-tracking branch 'origin/add-gradle-support'

Commit:
Add Gradle

Commit:
Add JUnit Tests

Commit:
Create a JAR File

Commit:
JavaDoc

Commit:
Follow the Coding Standard

Commit:
Find

Commit:
to avoid fast forwarding

Commit:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	.idea/workspace.xml

Commit:
no change

Commit:
Merge branch 'branch-Level-9'

# Conflicts:
#	.idea/workspace.xml
#	src/main/java/Judy/task/TaskList.java
#	src/main/java/Judy/ui/Parser.java

Commit:
Use CheckStyle

andong0909 commented on src/main/java/Judy/command/DeleteCommand.java:
```diff
@@ -0,0 +1,16 @@
+import Judy.ui.*;
+import Judy.util.*;
+
+public class DeleteCommand extends Command {
+    private final int index;
+
+    public DeleteCommand(int index) {
+        this.index = index;
+    }
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws JudyException {
+        tasks.deleteTask(index);
+    }
+}
```
Maybe it is easier to understand if you set the name of similar class/methods in similar forms? AddTaskCommand vs Delete(Task)Command, especially when you have very similar methods in different classes, eg, in Command and TaskList

andong0909 commented on src/main/java/Judy/command/FindCommand.java:
```diff
@@ -0,0 +1,17 @@
+import Judy.ui.*;
+import Judy.util.*;
+
+public class FindCommand extends Command{
+
+    private final String keyward;
+
+    public FindCommand(String keyward) {
+        this.keyward = keyward;
+    }
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws JudyException {
+        tasks.findTask(keyward);
+    }
+}
```
Maybe it is easier to understand if you set the name of similar class/methods in similar forms?

andong0909 commented on src/main/java/Judy/task/Todo.java:
```diff
@@ -0,0 +1,16 @@
+        super(description);
+    }
+
+    @Override
+    public String toString() {
+        return "[T]" + super.toString();
+    }
+
+    @Override
+    public String toDataString() {
+        return "T | " + super.toDataString();
+    }
```
Good job on clear naming!

andong0909 reviewed (commented):
Generally well done!

sreelakshmiharidos commented on src/main/java/Judy/task/TaskList.java:
```diff
@@ -0,0 +1,140 @@
+        return this.list;
+    }
+
+    /**
```
Great job on adding appropriate Javadoc comments!

sreelakshmiharidos commented on src/main/java/Judy/task/TaskList.java:
```diff
@@ -0,0 +1,140 @@
+            response.append("    ").append(index).append(". ").append(task.toString()).append("\n");
+            index++;
+        }
+
```
Perhaps you don't need a empty-line separation here? Removing it could improve readability while keeping related logic together!

sreelakshmiharidos commented on src/main/java/Judy/ui/Parser.java:
```diff
@@ -0,0 +1,72 @@
+            String[] parts = input.split("/from |/to ");
+            if (parts.length == 3) {
+                String[] description = parts[0].trim().split(" ", 2);
+                return new AddTaskCommand(new String[]{description[1]}, TaskType.EVENT, null, new String[]{parts[1].trim()}, new String[]{parts[2]});
```
Perhaps add a line break to improve readability?

sreelakshmiharidos commented on src/test/java/Judy/TaskTest.java:
```diff
@@ -0,0 +1,17 @@
+package Judy;
+import Judy.task.*;
+import org.junit.jupiter.api.Test;
```
Perhaps add a free line between package declaration and import statements to improve readability?

sreelakshmiharidos reviewed (commented):
LGTM. Just a few nits to improve readability!

Commit:
Add JavaFX GUI

Commit:
Add Java varargs feature to addTask

Commit:
Edit JavaFX GUI

Commit:
Edit JavaFX GUI

Commit:
Merge branch 'branch-Level-10'

# Conflicts:
#	.idea/workspace.xml
#	data/judy.txt

Commit:
Generate JAR file using Gradle shadow plugin

The Gradle `shadowJar` task is used to create a fat JAR (Uber JAR) that includes all dependencies.
This ensures that the generated JAR can run independently without requiring external dependencies.

The JAR file is placed in `build/libs/` and contains all necessary classes and resources.
Running `./gradlew clean shadowJar` ensures a fresh build.

This allows for easier deployment and execution of the application as a standalone JAR.

Commit:
Introduce a minor update to enable PR creation

Currently, GitHub prevents creating a pull request because the branch is identical to master.
To bypass this restriction, a minor change has been introduced. This ensures the PR can be
created while maintaining the integrity of the branch.

The change is minimal and does not impact functionality, serving only as a workaround
to allow the PR process to proceed.

Once the PR is successfully created and merged, this commit can be reverted if necessary.

ki1r0 committed:
Merge pull request #2 from ki1r0/branch-A-Assertions

PR - branch a assertions

Commit:
Revert "Merge pull request #2 from ki1r0/branch-A-Assertions"

This reverts commit 3bafea9dd08109d602f6dd6637353e841ca3c89d, reversing
changes made to 96371e8190b92fff0c1bc6f65ddf5e2df479db61.

ki1r0 committed:
Merge pull request #3 from ki1r0/branch-A-Code-Quality

PR: branch-A-CodeQuality

Commit:
Improve code quality by updating .gitignore and standardizing package naming

1. **Move binary files to .gitignore**
   - Added binary and generated files to `.gitignore` to prevent unnecessary files from being tracked.
   - This improves repository cleanliness and reduces clutter in version control.

2. **Standardize Package Name Style**
   - Updated package names to follow standard Java naming conventions.
   - Ensures consistency and improves code readability and maintainability.

These changes enhance the project's organization, making it easier to manage and collaborate efficiently.

Commit:
Introduce a minor update to enable PR creation

Currently, GitHub prevents creating a pull request because the branch is identical to master.
To bypass this restriction, a minor change has been introduced. This ensures the PR can be
created while maintaining the integrity of the branch.

The change is minimal and does not impact functionality, serving only as a workaround
to allow the PR process to proceed.

Once the PR is successfully created and merged, this commit can be reverted if necessary.

Commit:
Add assertions for input validation in command parsing

Added `assert` statements to ensure non-null input, valid command formats,
and proper parameter handling for `todo`, `deadline`, `event`, `mark`, and
`delete` commands. These changes improve debugging, enhance code robustness,
and prevent unexpected errors during execution.

ki1r0 committed:
Merge pull request #4 from ki1r0/branch-A-Assertions

Add assertions for input validation in command parsing

Commit:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/Judy/ui/Parser.java

Commit:
Merge branch 'master' into branch-A-Streams

Commit:
Merge branch 'master' into branch-A-Streams

ki1r0 committed:
Merge pull request #5 from ki1r0/branch-A-CodeQuality

Improve code quality

ki1r0 committed:
Merge pull request #6 from ki1r0/branch-A-Streams

Introduce a minor update to enable PR creation

Commit:
Current date parsing logic is inconsistent and lacks proper error handling.

The method fails to correctly determine whether a date format includes time, and it does not handle invalid inputs gracefully. Additionally, the day abbreviation matching logic is unnecessarily complex.

Improve date parsing reliability by checking for successful parsing instead of relying on formatter string properties. Simplify day abbreviation matching using startsWith(). Enhance error handling by explicitly catching exceptions and returning "Invalid date" for unrecognized inputs.

This approach ensures better maintainability, improves readability, and prevents unintended behavior when incorrect inputs are provided. The changes make the method more robust and user-friendly.

Commit:
Docs: Add user guide for system usage and best practices

Currently, new users may struggle to understand the system's
functionality due to the lack of clear documentation.

A user guide helps onboard users by providing structured explanations
of features, workflows, and best practices, reducing the learning
curve and improving the user experience.

Let's introduce a detailed user guide covering:
* System setup and installation steps
* Core features and how to use them effectively
* Troubleshooting common issues

This documentation enhances usability and reduces support requests
by offering self-service guidance.

Commit:
Docs: Update Ui.png with corrected application screenshot

The previous Ui.png contained errors, which could mislead users
regarding the actual appearance and functionality of the application.

An accurate and up-to-date screenshot ensures that users have a
correct visual reference when interacting with the application.

Let's replace the outdated Ui.png with a new version that reflects
the current UI correctly, improving clarity and consistency in
documentation.

Commit:
Docs: Update README.md for correct GitHub rendering

The previous README.md did not display correctly on the GitHub page,
potentially due to incorrect Markdown formatting or missing elements.

A properly formatted README enhances project visibility, making it
easier for users to understand the purpose, installation steps, and
usage of the application.

Let's update the README.md to ensure correct rendering on GitHub by:
* Fixing Markdown syntax issues
* Adjusting formatting for better readability
* Verifying proper display of images and links

This update improves documentation clarity and enhances the user experience.

Commit:
Code quality: Improve coding standards with Javadocs and import fixes

Some parts of the codebase lacked proper Javadoc comments, making it
harder to understand the purpose and usage of certain classes and methods.
Additionally, some imports were disorganized, affecting readability
and maintainability.

Well-documented code improves maintainability and helps new contributors
quickly grasp the code structure. Properly arranged imports enhance
readability and prevent potential conflicts.

Let's improve the coding standards by:
* Completing Javadocs for public classes and methods
* Rearranging problematic imports to follow best practices

These changes improve code clarity, maintainability, and adherence to
coding standards.

Commit:
Docs: Update README.md introduction for clarity and consistency

The previous README.md introduction lacked clarity and did not fully
explain the chatbot’s purpose, features, and functionality. This could
make it difficult for new users to understand the project's scope
and capabilities.

A well-structured introduction improves readability, helps new users
quickly grasp what the chatbot does, and enhances the overall
documentation quality.

Let's update the introduction to clearly define Judy Chatbot’s purpose,
list its key features, and provide a concise overview of its capabilities.

This change improves the documentation’s clarity, making it more user-friendly
and informative while maintaining a professional tone.

Commit:
Build/UI: Create fat JAR and fix JavaFX error message display

Previously, error messages were not displaying correctly in the JavaFX UI,
making debugging difficult. Additionally, the application did not have a
properly packaged fat JAR, complicating deployment by requiring manual
dependency management.

A fat JAR simplifies distribution by bundling all dependencies into a
single executable file. Fixing the JavaFX error message display ensures
that errors are properly visible to users and developers.

Let's:
* Generate a fat JAR to streamline application deployment
* Debug and fix JavaFX error handling so error messages display correctly
* Ensure error dialogs or logs provide meaningful feedback

This improves deployment efficiency and enhances debugging by making
errors visible within the JavaFX interface.

ki1r0 committed:
Docs: Update README.md with correct release link  Previously, the README.md contained an incorrect or outdated release link,  which could mislead users when downloading the application.  Providing the correct release link ensures that users can easily access  the latest version without confusion.  Let's update the README.md to include the correct release link, improving  the accuracy and usability of the documentation.

ki1r0 committed:
Docs: Update README.md with correct release link  Previously, the README.md contained an incorrect or outdated release link,  which could mislead users when downloading the application.  Providing the correct release link ensures that users can easily access  the latest version without confusion.  Let's update the README.md to include the correct release link, improving  the accuracy and usability of the documentation.

Commit:
Bug fix: Prevent StringIndexOutOfBoundsException for empty todo descriptions

Previously, entering a `todo` command without a description caused a
`StringIndexOutOfBoundsException`, as the code attempted to access
characters beyond the valid index range.

An unhandled exception can cause unexpected crashes and poor user
experience. Ensuring proper validation prevents such issues.

Let's fix this by:
* Adding a check to detect empty `todo` descriptions before processing
* Displaying a meaningful error message instead of throwing an exception

This improves stability and ensures the application handles invalid
input gracefully.

Commit:
Refactor: Improve error handling for consistency and clarity

Previously, error messages were inconsistent and sometimes lacked
sufficient detail, making it harder for users to understand what went
wrong and how to fix it.

Clear and standardized error messages improve user experience and
debugging efficiency by providing meaningful feedback.

Let's optimize error handling by:
* Standardizing error messages across all commands
* Providing more informative details about the issue
* Ensuring errors are displayed in a user-friendly format

This refactor enhances usability, improves debugging, and makes the
application more robust against invalid inputs.

Commit:
UI: Enable automatic resizing of interface components

Currently, the interface components do not resize dynamically when the
window is resized. This results in a poor user experience, as elements
may appear misaligned or not fully visible on different window sizes.

A responsive UI improves usability by ensuring that all components
adjust proportionally to window size changes.

Let's enable automatic resizing for interface components by setting
appropriate layout constraints and using responsive design principles.

This improves visual consistency and ensures a smoother user experience
across different screen sizes.

Commit:
Parser: Fix StringIndexOutOfBoundsException in event parsing

Parsing an event task without a valid description or missing parameters
caused a StringIndexOutOfBoundsException.

Fix this by adding input validation to prevent invalid substring access.

Commit:
Refactor: Fix typos in variable names for clarity

Some variable names contained typos, which could lead to confusion and
reduce code readability.

Fix this by correcting the typos to ensure consistency and maintainability.

Commit:
Refactor: Remove unused Day class

The Day class is no longer used in the codebase, making it redundant
and unnecessary.

Remove it to improve code maintainability and reduce clutter.
